Tests/IntegrationTests.Shared/PrimaryKeyTests.cs

```diff
@@ -0,0 +1,201 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using NUnit.Framework;
+using System.Threading;
+using Realms;
+
+namespace IntegrationTests.Shared
+{
+    // using classes from TestObjects.cs
+    [TestFixture, Preserve(AllMembers = true)]
+    public class PrimaryKeyTests
+    {
+
+        protected Realm _realm;
+
+        [SetUp]
+        public void Setup()
+        {
+            Realm.DeleteRealm(RealmConfiguration.DefaultConfiguration);
+            _realm = Realm.GetInstance();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _realm.Close();
+            Realm.DeleteRealm(_realm.Config);
+        }
+
+        [Test]
+        public void FindByCharPrimaryKey()
+        {
+            _realm.Write(() => {
+                var obj = _realm.CreateObject<PrimaryKeyCharObject>();
+                obj.CharProperty = 'x';
+            });
+
+            var foundObj = _realm.ObjectForPrimaryKey<PrimaryKeyCharObject>('x');
+            Assert.IsNotNull(foundObj);
+            Assert.That(foundObj.CharProperty, Is.EqualTo('x'));
```

Comment:
If `obj` was declared in the outer scope, we could assert that `foundObj` and `obj` are equal instead.